pydantic_ai_slim/pydantic_ai/providers/gateway.py

```diff
@@ -73,22 +82,38 @@ def gateway_provider(
 @overload
 def gateway_provider(
     upstream_provider: str,
+    /,
     *,
     api_key: str | None = None,
     base_url: str | None = None,
 ) -> Provider[Any]: ...
 
 
-UpstreamProvider = Literal['openai', 'openai-chat', 'openai-responses', 'groq', 'google-vertex', 'anthropic', 'bedrock']
+UpstreamProvider = Literal[
+    'chat',
+    'responses',
+    'gemini',
+    'converse',
+    # Deprecated
+    'openai',
+    'openai-chat',
+    'openai-responses',
+    'google-vertex',
+    'bedrock',
+    # Those two are actually the same on both.
+    'anthropic',
+    'groq',
+]
 
 
 def gateway_provider(
     upstream_provider: UpstreamProvider | str,
+    /,
```

Comment:
This is a breaking change!!!

We do not document `gateway_provider` - I would like to follow the "if it's not documented, it's not public" approach.

---

We could also rename `upstream_provider` to `api_type` here.

---

If we are not okay with this, then the annoying path to deprecation it is...